ros2pkg/package.xml

```diff
@@ -12,6 +12,8 @@
   <depend>ros2cli</depend>
 
   <exec_depend>ament_index_python</exec_depend>
+  <exec_depend>python3-empy</exec_depend>
+  <exec_depend>python3-setuptools</exec_depend>
```

Comment:
I don't think we declare (in the package.xml) dependency on setuptools for pure python packages (the same way we don't declare a dependency on cmake for ament_cmake packages).

Not sure about listing it in the install_requires section of the setup.py, it seems that some packages do and some don't. Maybe @dirk-thomas can give more details on that point

Reply:
I based my comment on the content of this repository. the `ros2cli` package (dependency of this one) uses `setuptools` and `pkg_resources` in multiple locations and doesnt declare a dependency on it in either the `package.xml` or the python files that import the module. 
Is the general rule that any package using pkg_resources should declare that dependency in their package.xml ?

Reply:
Yes, that would make sense.

Reply:
I guess this should be python3-pkg-resources then

Reply:
Keys used in `package.xml` are used to query the rosdep database. In general we try to make sure that the rosdep keys resolve to apt packages if such package exist and avoid pip rules as much as we can as we cannot embed these dependencies in our debs.

My guess is that is was an oversight when the [dependency was added to ament_package](https://github.com/ament/ament_package/pull/50) to fix failing downstream packages. As [setuptools depends on pkg_resources](https://packages.ubuntu.com/xenial/python3-setuptools), the more generic dependency does fix the problem.

Reply:
I would guess the dependency was even added in to the xml before the rosdep rules existed. Any the assumption was probably at that time there is only `setuptools` - as you said `pkg-resources` is not a PyPI package.

Reply:
ok: minimal, apt-favouring dependency of `python3-pkg-resources` used in https://github.com/ros2/ros2cli/pull/87/commits/7c2aef86b21be98700f93ee7ea7398de3604b1ea :+1: